tests/commands/test_store_client.py

```diff
@@ -213,11 +219,12 @@ def test_authholder_request_simple(auth_holder):
 
     other_cookie = get_cookie(value='different')
 
-    def fake_request(self, method, url, json):
+    def fake_request(self, method, url, json, headers):
         # check it was properly called
         assert method == 'testmethod'
         assert url == 'testurl'
         assert json == 'testbody'
+        assert headers == {'User-Agent': USER_AGENT}
```

Comment:
I do understand Don't Repeat Yourself.
I worry a bit about tests that do this, and become sort of tautologies vs having a useful "the user agent string is something that is a nice useful value"
assert headers == {'User-Agent': 'charmcraft/0.1.0'}

I'm a bit mixed here, but the above test leads to user agent strings that humans never see and you don't realize they've gotten out of hand and aren't properly meaningful.

We don't have to change this, but we should discuss it a bit and decide.

Reply:
I think those tests should just use the user agent without worrying their format (as the test asserts that it's *using* the UA, nothing else).

That said, you have a point about the format of the UA itself, and never seeing "the final form". That's why I added a test specifically for the UA, which currently is very simple but may grow in complexity when we solve #74 (had to stop using a module constant in favor of a function, but it's fine).